### PR TITLE
`reflect`: Select constants and stubs

### PR DESCRIPTION
**Pull request by bgould**
Adds constants and stubs to `reflect` package to support compilation mentioned in https://github.com/tinygo-org/tinygo/issues/742#issuecomment-1411026796

### CODE REVIEW

**Review by dgryski — CHANGES_REQUESTED** [review]
These should probably be in `value.go` with everything else. I just put them in chan.go for convenience when I was trying to get the lua stuff to build.

File: src/reflect/chan.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	Send Value     // value to send (for send)
+}
+
+func Select(cases []SelectCase) (chosen int, recv Value, recvOK bool) {
+	panic("reflect.Select: unimplemented")
+}
+
+func (v Value) Send(x Value) {
+	panic("reflect.Value.Send(): unimplemented")
+
```
Comment (aykevl) [inline src/reflect/chan.go:24]: Nit: unnecessary newline

File: src/reflect/chan.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+func (v Value) Close() {
+	panic("reflect.Value.Close(): unimplemented")
+
```
Comment (aykevl) [inline src/reflect/chan.go:29]: Same here.

**Review by aykevl — COMMENTED** [review]
> These should probably be in `value.go` with everything else.

Agreed, plus two really minor nits you can fix at the same time. Other than that, this looks good to me.

**Comment by dgryski** [comment]
Incorporated this change into https://github.com/tinygo-org/tinygo/pull/3470

**Pull request closed without merging**